qpc/request.py

```diff
@@ -169,58 +169,86 @@ def request(
-            )
-        elif method == GET:
-            result = handle_general_errors(
-                get(url, params, req_headers), min_server_version
-            )
-        elif method == PATCH:
-            result = handle_general_errors(
-                patch(url, payload, req_headers), min_server_version
-            )
-        elif method == DELETE:
-            result = handle_general_errors(delete(url, req_headers), min_server_version)
-        elif method == PUT:
-            result = handle_general_errors(
-                put(url, payload, req_headers), min_server_version
+        methods = {
+            "POST": post,
+            "GET": get,
+            "PATCH": patch,
+            "DELETE": delete,
+            "PUT": put
+        }
+        try:
+            request_method = methods[method]
+            result = perform_request(
+                method,
+                request_method,
+                url,
+                params,
+                payload,
+                req_headers,
+                min_server_version
             )
-        else:
+        except KeyError:
             logger.error("Unsupported request method %s", method)
             parser.print_help()
             sys.exit(1)
-        try:
-            log_request_info(
-                method, log_command, url, result.json(), result.status_code
-            )
-        except ValueError:
-            log_request_info(method, log_command, url, result.text, result.status_code)
+        log_request_info(
+            method, log_command, url, decode_response_json(result), result.status_code
+        )
         return result
     except (requests.exceptions.ConnectionError, requests.exceptions.SSLError):
-        config = read_server_config()
-        if config is not None:
-            protocol = "https"
-            host = config.get(CONFIG_HOST_KEY)
-            port = config.get(CONFIG_PORT_KEY)
-            if config.get(CONFIG_USE_HTTP):
-                protocol = "http"
-            logger.error(
-                _(CONNECTION_ERROR_MSG),
-                {"protocol": protocol, "host": host, "port": port}
-            )
-            logger.error(_(messages.SERVER_CONFIG_REQUIRED), PKG_NAME)
-        else:
-            logger.error(_(messages.SERVER_CONFIG_REQUIRED), PKG_NAME)
+        handle_connection_error()
         sys.exit(1)
+
+
+def handle_connection_error():
+    """Log connection error."""
+    config = read_server_config()
+    if config is not None:
+        protocol = "https"
+        host = config.get(CONFIG_HOST_KEY)
+        port = config.get(CONFIG_PORT_KEY)
+        if config.get(CONFIG_USE_HTTP):
+            protocol = "http"
+        logger.error(
+            _(CONNECTION_ERROR_MSG),
+            {"protocol": protocol, "host": host, "port": port}
+        )
+    logger.error(_(messages.SERVER_CONFIG_REQUIRED), PKG_NAME)
+
+
+def perform_request(
+    method,
+    request_method,
```

Comment:
Hmm…

It feels weird that `request` is using `method` to get `request_method`, and then it just passes _both_ of those into `perform_request`, and `request` does nothing else with the `request_method` that it just created. This makes me think there's no real benefit to instantiating `request_method` outside of `perform_request`, and `request` should send just `method` to `perform_request`, and `perform_request` should do the `request_method` lookup.

Beyond the scope of this change (and I am _not_ suggesting you do the following in _this_ PR), I would like to strip away some unnecessary layers of abstraction in this module. Our current `get`, `post`, etc. functions are doing nothing useful. They just add complication to functions like this. 😕

Reply:
Sorry if what I said there seems incoherent. I'm not sure if I clearly expressed my confusing line of thoughts. 🤔 

I want to revisit all this code _in general_ some time later, not just what you've written in this PR, to straighten out the seemingly useless `get`, `post`, etc. functions, and I expect most of `request_method` would disappear if/when that happens. So, my suggestion to move the `request_method` lookup from `request` to `request_method` would technically be an improvement, but it's a _very minor_ one, and not something that should block the PR in its current state.

Reply:
OK, I understand. I will work on a better solution applying your suggestions and push a fixup.